ACE/ace/OS_NS_Thread.cpp

```diff
@@ -4004,6 +4006,18 @@ ACE_OS::thr_create (ACE_THR_FUNC func,
 
 #   endif /* sun && ACE_HAS_ONLY_SCHED_OTHER */
   auto_thread_args.release ();
+
+  // *** Set pthread name (second try)
+#     if defined (ACE_HAS_PTHREAD_SETNAME_NP)
+  if (!name_set && thr_name && *thr_name)
```

Comment:
Can the bool not be removed when you only do this code when ACE_HAS_PTHREAD_ATTR_SETNAME is not defined?